fastly/block_fastly_waf_configuration_v1_exclusion.go

```diff
@@ -112,20 +112,25 @@ func updateWAFRuleExclusions(d *schema.ResourceData, meta interface{}, wafID str
 		ns = new(schema.Set)
 	}
 
-	oss := os.(*schema.Set)
-	nss := ns.(*schema.Set)
+	oldSet := os.(*schema.Set)
+	newSet := ns.(*schema.Set)
 
-	add := nss.Difference(oss).List()
-	remove := oss.Difference(nss).List()
+	setDiff := NewSetDiff(func(resource interface{}) (interface{}, error) {
+		// Use the resource name as the key
+		return resource.(map[string]interface{})["name"], nil
```

Comment:
This is the "name" attribute within the top level `wafRuleExclusion` resource which is what I _think_ is required. 
The reason I selected that as the key is because just before this code is `d.GetChange("rule_exclusion")`.
The `rule_exclusion` attribute exists in a different file that assigns to it `wafRuleExclusion`.
@phamann you might be best to sanity check me here.